p2p/starknet/client.go

```diff
@@ -0,0 +1,129 @@
+package p2p
+
+import (
+	"context"
+
+	p2p_spec "github.com/NethermindEth/juno/p2p/starknet/p2p-spec"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/protocol"
+	"google.golang.org/protobuf/proto"
+)
+
+type NewStreamFunc func(ctx context.Context, pids ...protocol.ID) (network.Stream, error)
+
+type StarknetClient struct {
```

Comment:
`starknet.StarknetClient` stutters. I think `Client` should suffice. We should rename the `NewStarknetClient` function to `NewClient` as well I think.

Reply:
We are still in the `p2p` package, I just wanted to have some level of separation between generic p2p and starknet specific stuff. Hence the `starknet` subfolder. Is there any better way of doing this, I am open to suggestions?

Reply:
> We are still in the p2p package

I missed that, thanks.

> I just wanted to have some level of separation between generic p2p and starknet specific stuff

I feel the same.

When I import `github.com/NethermindEth/juno/p2p/starknet` in `node/node.go`, I get an error saying that the package name clashes with the `p2p` package. Since everyone who imports both will have to alias the import anyway, maybe its worth renaming the package to `starknet`?

Reply:
Ah, I didnt realize that. `starknet` sounds good.

Reply:
done